qiskit_nature/algorithms/pes_samplers/bopes_sampler.py

```diff
@@ -24,29 +24,62 @@
     ElectronicStructureMoleculeDriver,
     VibrationalStructureMoleculeDriver,
 )
+from qiskit_nature.converters.second_quantization import QubitConverter
 from qiskit_nature.exceptions import QiskitNatureError
 from qiskit_nature.problems.second_quantization import BaseProblem
 from qiskit_nature.results import BOPESSamplerResult, EigenstateResult
 from ..ground_state_solvers import GroundStateSolver
 from .extrapolator import Extrapolator, WindowExtrapolator
+from ..ground_state_solvers.minimum_eigensolver_factories import MinimumEigensolverFactory
 
 logger = logging.getLogger(__name__)
 
 
+def prepare_problem(problem: BaseProblem, qubit_converter: QubitConverter):
+    """Prepare raw problem, make necessary adjustments for the computation.
+    Args:
+        problem: a class encoding a problem to be solved.
+        qubit_converter: Qubit converter
+    Returns:
+        problem: a class encoding a problem to be solved.
+    Raises:
+        ValueError: if the grouped property object returned by the driver does not contain a
+            main property as requested by the problem being solved (`problem.main_property_name`)
+    """
+    second_q_ops = problem.second_q_ops()
+    if isinstance(second_q_ops, list):
+        main_second_q_op = second_q_ops[0]
+    elif isinstance(second_q_ops, dict):
+        name = problem.main_property_name
+        main_second_q_op = second_q_ops.pop(name, None)
+        if main_second_q_op is None:
+            raise ValueError(
+                f"The main `SecondQuantizedOp` associated with the {name} property cannot be "
+                "`None`."
+            )
+    main_operator = qubit_converter.convert(
+        main_second_q_op,
+        num_particles=problem.num_particles,
+        sector_locator=problem.symmetry_sector_locator,
+    )
+    return main_operator
+
+
 class BOPESSampler:
     """Class to evaluate the Born-Oppenheimer Potential Energy Surface (BOPES)."""
 
     def __init__(
         self,
-        gss: GroundStateSolver,
+        gss: Union[GroundStateSolver, MinimumEigensolverFactory],
         tolerance: float = 1e-3,
         bootstrap: bool = True,
         num_bootstrap: Optional[int] = None,
         extrapolator: Optional[Extrapolator] = None,
+        qubit_converter: Optional[QubitConverter] = None,
     ) -> None:
         """
         Args:
-            gss: GroundStateSolver
+            gss: GroundStateSolver or MinimumEigensolverFactory
```

Comment:
I suggest you mention that in the latter case a `QubitConverter` is required already here.